src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,782 @@
+            public string Prefix;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Columns to drop after the DateTime Expansion", Name = "ColumnsToDrop", ShortName = "drop", SortOrder = 3)]
+            public ColumnsProduced[] ColumnsToDrop;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Country to get holidays for. Defaults to none if not passed", Name = "Country", ShortName = "ctry", SortOrder = 4)]
+            public Countries Country = Countries.None;
+        }
+
+        #endregion
+
+        public DateTimeTransformerEstimator(IHostEnvironment env, string inputColumnName, string columnPrefix, ColumnsProduced[] columnsToDrop, Countries country = Countries.None)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+            _host.CheckValue(inputColumnName, nameof(inputColumnName), "Input column should not be null.");
+
+            _options = new Options
+            {
+                Source = inputColumnName,
+                Prefix = columnPrefix,
+                ColumnsToDrop = columnsToDrop == null ? Array.Empty<ColumnsProduced>() : columnsToDrop,
+                Country = country
+            };
+        }
+
+        internal DateTimeTransformerEstimator(IHostEnvironment env, Options options)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+
+            _options = options;
+            _options.ColumnsToDrop = _options.ColumnsToDrop == null ? Array.Empty<ColumnsProduced>() : _options.ColumnsToDrop;
+        }
+
+        public DateTimeTransformer Fit(IDataView input)
+        {
+            return new DateTimeTransformer(_host, _options.Source, _options.Prefix, _options.ColumnsToDrop, _options.Country);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (ColumnsProduced column in Enum.GetValues(typeof(ColumnsProduced)))
+                if (_options.ColumnsToDrop == null || !_options.ColumnsToDrop.Contains(column))
+                {
+                    columns[_options.Prefix + column.ToString()] = new SchemaShape.Column(_options.Prefix + column.ToString(), SchemaShape.Column.VectorKind.Scalar,
+                    ColumnTypeExtensions.PrimitiveTypeFromType(column.GetRawColumnType()), false, null);
+                }
+
+            return new SchemaShape(columns.Values);
+        }
+
+        #region Enums
+        public enum ColumnsProduced : byte
+        {
+            Year = 1, Month, Day, Hour, Minute, Second, AmPm, Hour12, DayOfWeek, DayOfQuarter, DayOfYear,
+            WeekOfMonth, QuarterOfYear, HalfOfYear, WeekIso, YearIso, MonthLabel, AmPmLabel, DayOfWeekLabel,
+            HolidayName, IsPaidTimeOff
+        };
```

Comment:
Recommend explicitly numbering each enum name. This lets us insert new values later without affecting the enum numeric value. 

This is more important if the enum is serialized within the model, which I expect you're doing since the model needs to know which features to produce.

Reply:
Good point. They are now explicitly numbered. The `Countries` enum (now `HolidayList`) is also numbered now.

---
In reply to: [353910235](https://github.com/dotnet/machinelearning/pull/4521#discussion_r353910235) [](ancestors = 353910235)